bench/benchmarks/fps.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+var Evented = require('../../js/util/evented');
+var util = require('../../js/util/util');
+var formatNumber = require('../lib/format_number');
+var measureFramerate = require('../lib/measure_framerate');
+
+var smallFeatureCollection = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "properties": {},
+        "geometry": {
+            "type": "Point",
+            "coordinates": [ -77.032194, 38.912753 ]
+        }
+    }]
+};
+
+var largeFeatureCollection = require('../data/naturalearth-land.json');
+
+var DURATION_MILLISECONDS = 5 * 1000;
+
+module.exports = function(options) {
+    var evented = util.extend({}, Evented);
+
+    var map = options.createMap({
+        width: 1024,
+        height: 768,
+        zoom: 5,
+        center: [-77.032194, 38.912753],
+        style: 'mapbox://styles/mapbox/bright-v8'
+    });
+
+    map.on('load', function() {
+        map.addSource('geojson', {
+            'type': 'geojson',
+            'data': {
+                'type': 'FeatureCollection',
+                'features': []
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-polygon',
+            'source': 'geojson',
+            'type': 'fill',
+            'filter': ['all', ['==', '$type', 'Polygon']],
+            'paint': {
+                'fill-color': '#000000',
+                'fill-opacity': 0.25
+            }
+        });
+
+        map.addLayer({
+            'id': 'geojson-point',
+            'source': 'geojson',
+            'type': 'circle',
+            'filter': ['all', ['==', '$type', 'Point']],
+            'paint': {
+                'circle-radius': 5,
+                'circle-color': '#000000'
+            }
+        });
+
+        var source = map.getSource('geojson');
+
+        var testSetData = function(geojson, cb) {
+            var running = true;
+            requestAnimationFrame(function runner() {
+                if (running) {
+                    source.setData(geojson);
+                    requestAnimationFrame(runner);
+                }
+            });
+
+            measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
+                running = false;
+                cb(err, fps);
+            });
+        };
+
+        map.repaint = true;
+        measureFramerate(DURATION_MILLISECONDS, function(err, fps) {
```

Comment:
Cool. Breaking these up. I put them together as seeing the normal fps was helpful, but I agree this limits the FPS benchmark from getter better.